sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs

```diff
@@ -110,13 +66,18 @@ public virtual async Task<Response<DataFeed>> GetDataFeedAsync(string dataFeedId
         {
             Guid dataFeedGuid = ClientCommon.ValidateGuid(dataFeedId, nameof(dataFeedId));
 
-            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsAdvisorAdministrationClient)}.{nameof(GetDataFeed)}");
+            using DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(MetricsAdvisorAdministrationClient)}.{nameof(GetDataFeed)}");
```

Comment:
We are also creating a DiagnosticScope inside GetDataFeedByIdAsync method.  I believe we should only have one -- for simplicity I suggest removing the outer one.

Reply:
Just curious, is it possible that something can fail during serialization(Input Model -> RequestContent) or deserialization(Response -> Output Model) and we need DiagnosticScope to trace the issue in grow up method?

Reply:
Hi @lmolkova, ​ I've a question, we are creating DiagnosticScope in the outer method as well as in the inner method. Outer method is handwritten and we do serialization and deserialization here and inner call is generated by C# generator and it is responsible for actual http call to the service. Both outer and inner methods are public. Is it fine to create DiagnosticScope in outer call as well as in the inner call?

Reply:
Summarizing offline conversation with @ShivangiReja:
- having both outer and inner scopes does not add a lot of value: duration and outcome are likely the same. 
- If there is a problem with serialization (i.e. outer scope is failed), users would be able to see exceptions and understand the problem is with serialization only.
- having two scopes increases telemetry costs for users 

Based on this, I suggest suppressing inner scope conditionally (when there is outer scope) and keeping it when inner method is called by users directly.

How to detect the presence of outer scope:

- Option 1: Cheap, hacky, sub-optimal. Check if the current Activity is created by the same DiagnosticScopeFactory using something like `bool hasOuterScope = Activity.Current?.Tags.Any(t => t.Key == "az.namespace" && t.Value == "<AzureResourceProviderNamespace>")`

- Option 2: A bit more complicated, but more generic and efficient: implement suppression logic inside `ClientDiagnostics`\`DiagnosticScope` using either `Activity.SetCustomProperty("<sdk-name>-scope-started", true)` or `AsyncLocal` field indicating that outer scope has started.

- Option 3: Do nothing and keep inner and outer scopes. We don't have suppression logic in other SDKs (even when public methods call into other public methods). But it seems we'll have more and more of such problems with LLC and I'd prefer to find a solution.
 
Option 1 sounds reasonable short-term solution and I can experiment with Option 2 in a couple of weeks. It should be possible to switch to Option 2 after. Thoughts?

Reply:
We're not in urgent need of this, so let's go with Option 3 while we work out a nicer solution.  I've created https://github.com/Azure/autorest.csharp/issues/2069 to keep track of using the Option 2 solution.

@lmolkova, do we consider changes to telemetry a breaking change?  If we do we'll want to make sure we have this before the DPG GA.  But I expect reducing telemetry costs would instead be a welcome change to customers.

Reply:
@annelo-msft Live tests are failing because for some methods we are creating outer and inner scopes with the same name.
For example:
**Outer scope**: [MetricsAdvisorAdministrationClient.CreateDataFeed](https://github.com/ShivangiReja/azure-sdk-for-net/blob/MetricsAdvisor-Experiment/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs#L278)
**Inner scope**: [MetricsAdvisorAdministrationClient.CreateDataFeed](https://github.com/ShivangiReja/azure-sdk-for-net/blob/MetricsAdvisor-Experiment/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Generated/MetricsAdvisorAdministrationClient.cs#L1770)


**Live test failure**:

![image](https://user-images.githubusercontent.com/45216704/158493549-f0939e56-130d-40af-abe5-00e97805540d.png)


**Live test failure in the pipeline**:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=1436751&view=logs&j=df47e0b9-f7f0-5a6d-001b-395452402a29&t=a918488f-3fbc-5b68-faba-d1917c3f3686&l=581

Reply:
To workaround this issue I'm removing outer scope created in grow-up method which is handwritten and responsible for serialization and deserialization until we implement https://github.com/azure/autorest.csharp/issues/2069 based @lmolkova's solution.

Reply:
@ShivangiReja sounds good!

> @lmolkova, do we consider changes to telemetry a breaking change? If we do we'll want to make sure we have this before the DPG GA. But I expect reducing telemetry costs would instead be a welcome change to customers.

@annelo-msft we don't have guidelines on it and I think we should decide on a case-by-case basis. With the current approach, we'd only have the inner scope, but then will replace it with outer after we have https://github.com/Azure/autorest.csharp/issues/2069, which is, in theory, breaking too. So it'd be ideal to resolve it before GA.